lib/active_model/serializer.rb

```diff
@@ -26,35 +26,12 @@ class << self
     end
 
     def self.inherited(base)
-      base._attributes = self._attributes.try(:dup) || []
-      base._attributes_keys = self._attributes_keys.try(:dup) || {}
```

Comment:
I know it's outside the scope of this PR, but when I'm looking at this code, I'm thinking _attributes and _attributes_keys should never be nil, so why not initialize them when defined?  Just curious if that makes any sense.

e.g.

```
class << self
      attr_accessor :_attributes
      attr_accessor :_attributes_keys
end
self._attributes = []
self._attributes_keys = {}
```

Or since we have active support, we might as well use the [class_attribute](https://github.com/rails/rails/blob/4-2-stable/activesupport/lib/active_support/core_ext/class/attribute.rb#L6-L75) macro and just

```ruby
require 'active_support/core_ext/class/attribute'
class_attribute :_attributes
Serializer._attributes ||= []
class_attribute :_attributes_keys
Serializer._attributes_keys = {}
```

or maybe just [`cattr_accessor`](https://github.com/rails/rails/blob/4-2-stable/activesupport/lib/active_support/core_ext/class/attribute_accessors.rb)